src/redisearch_rs/trie_rs/src/iter/range.rs

```diff
@@ -293,42 +293,42 @@ impl<'tm, Data> RangeIter<'tm, Data> {
                 self.stack.reserve(node.children().len());
 
                 let mut max_index = node.children().len();
-                if let Some(max) = child_max {
-                    if let Some(first) = max.first() {
-                        max_index = match node.children_first_bytes().binary_search(first) {
-                            Ok(i) => {
-                                self.stack.push(StackEntry {
-                                    node: &node.children()[i],
-                                    was_visited: false,
-                                    min: child_min,
-                                    max: child_max,
-                                });
-                                i
-                            }
-                            Err(i) => i,
-                        };
-                    }
+                if let Some(max) = child_max
+                    && let Some(first) = max.first()
+                {
+                    max_index = match node.children_first_bytes().binary_search(first) {
+                        Ok(i) => {
+                            self.stack.push(StackEntry {
+                                node: &node.children()[i],
+                                was_visited: false,
+                                min: child_min,
+                                max: child_max,
+                            });
+                            i
+                        }
+                        Err(i) => i,
+                    };
                 }
 
                 let mut min_index = 0;
 
                 let mut min_entry = None;
-                if let Some(min) = child_min {
-                    if let Some(first) = min.first() {
-                        min_index =
-                            match node.children_first_bytes()[..max_index].binary_search(first) {
-                                Ok(i) => {
-                                    min_entry = Some(StackEntry {
-                                        node: &node.children()[i],
-                                        was_visited: false,
-                                        min: child_min,
-                                        max: child_max,
-                                    });
-                                    i + 1
-                                }
-                                Err(i) => i,
-                            };
-                    }
+                if let Some(min) = child_min
+                    && let Some(first) = min.first()
```

Comment:
Maybe a `match` will be simpler?

Reply:
Since we only care about one branch, I'd say no.